services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/gdpr/KortistoGdprServiceITest.kt

```diff
@@ -319,12 +335,245 @@ class KortistoGdprServiceITest(
+                    hakija(kayttooikeustaso = Kayttooikeustaso.KAIKKI_OIKEUDET)
+                }
+
+            gdprService.deleteInfo(USERID)
+
+            assertThat(hankeRepository.findAll()).single().prop(HankeEntity::id).isEqualTo(hanke.id)
+            assertThat(applicationRepository.findAll())
+                .single()
+                .prop(ApplicationEntity::id)
+                .isEqualTo(hakemus.id)
+            assertThat(hankeKayttajaService.getKayttajatByHankeId(hanke.id)).single().all {
+                prop(HankeKayttajaDto::kayttooikeustaso).isEqualTo(Kayttooikeustaso.KAIKKI_OIKEUDET)
+                prop(HankeKayttajaDto::sukunimi).isEqualTo(KAYTTAJA_INPUT_HAKIJA.sukunimi)
+                prop(HankeKayttajaDto::id).isNotEqualTo(kayttaja.id)
+            }
+        }
+
+        @Test
+        fun `deletes hankekayttaja when the kayttaja is not an admin`() {
+            lateinit var kayttaja: HankekayttajaEntity
+            val hanke =
+                hankeFactory.builder("Other user").withHankealue().saveWithYhteystiedot {
+                    kayttaja =
+                        kayttaja(
+                            kayttooikeustaso = Kayttooikeustaso.HAKEMUSASIOINTI,
+                            userId = USERID
+                        )
+                    omistaja { addYhteyshenkilo(it, kayttaja) }
+                }
+            val hakemus =
+                hakemusFactory.builder(USERID, hanke).saveWithYhteystiedot {
+                    hakija { addYhteyshenkilo(it, kayttaja) }
+                }
+
+            gdprService.deleteInfo(USERID)
+
+            assertThat(hankeRepository.findAll()).single().prop(HankeEntity::id).isEqualTo(hanke.id)
+            assertThat(applicationRepository.findAll())
+                .single()
+                .prop(ApplicationEntity::id)
+                .isEqualTo(hakemus.id)
+            assertThat(hankeKayttajaService.getKayttajatByHankeId(hanke.id)).single().all {
+                prop(HankeKayttajaDto::kayttooikeustaso).isEqualTo(Kayttooikeustaso.KAIKKI_OIKEUDET)
+                prop(HankeKayttajaDto::sukunimi).isEqualTo(ProfiiliFactory.DEFAULT_NAMES.lastName)
+                prop(HankeKayttajaDto::id).isNotEqualTo(kayttaja.id)
+            }
+        }
+
+        @Test
+        fun `throws an exception when the kayttaja is on an active application`() {
+            val hanke =
+                hankeFactory.builder("Other user").withHankealue().saveWithYhteystiedot {
+                    omistaja()
+                }
+            val kayttaja = hankekayttajaFactory.saveIdentifiedUser(hanke.id, userId = USERID)
+            val hakemus =
+                hakemusFactory.builder(USERID, hanke).inHandling().saveWithYhteystiedot {
+                    hakija { addYhteyshenkilo(it, kayttaja) }
+                }
+
+            val failure = assertFailure { gdprService.deleteInfo(USERID) }
+
+            failure
```

Comment:
This failure is identical to the one thrown in test `throws and exception when the user is the only admin and there's an active application without the user` above.  Should it be somehow different?

Reply:
This can be ignored as you already commented on similar situation in PR #633